### PR TITLE
Optimize header includes (4/N)

### PR DESCRIPTION
**Pull request by mvieth**
As recommended by include-what-you-use
Continuing #4289 and others.

### CODE REVIEW

File: surface/src/on_nurbs/fitting_surface_pdm.cpp [hunk 1]
```diff
@@ -38,6 +38,7 @@
 #include <pcl/surface/on_nurbs/fitting_surface_pdm.h>
 #include <pcl/pcl_macros.h>
 
+#include <Eigen/Cholesky>
```
Comment (kunaltyagi) [inline surface/src/on_nurbs/fitting_surface_pdm.cpp:41]: Please add the `// for` recommended by IWYU in additions

Comment (mvieth) [reply surface/src/on_nurbs/fitting_surface_pdm.cpp]: Added, thanks.

File: visualization/src/interactor_style.cpp [hunk 2]
```diff
@@ -38,6 +38,7 @@
 
 #include <list>
 #include <pcl/common/angles.h>
+#include <pcl/visualization/boost.h>
```
Comment (kunaltyagi) [inline visualization/src/interactor_style.cpp:41]: Missing `// for` here

Comment (mvieth) [reply visualization/src/interactor_style.cpp]: Split into more specific includes and added explanation

**Review by SergioRAgostinho — APPROVED** [review]

**Review by kunaltyagi — APPROVED** [review]
Waiting on CI

**Pull request merged**